migration-engine/connectors/sql-migration-connector/src/sql_schema_differ.rs

```diff
@@ -61,48 +60,39 @@ pub(crate) fn calculate_steps(schemas: Pair<&SqlSchema>, flavour: &dyn SqlFlavou
-        //   might be named the same as the dropped enum, and that conflicts on
-        //   postgres.
-        // - We must drop enums after we drop tables, or dropping the enum will
-        //   fail on postgres because objects (=tables) still depend on them.
-        .chain(flavour.drop_enums(&differ).into_iter().map(SqlMigrationStep::DropEnum))
-        .chain(create_tables.into_iter().map(SqlMigrationStep::CreateTable))
-        .chain(redefine_tables)
-        // Order matters: we must create indexes after ALTER TABLEs because the indexes can be
-        // on fields that are dropped/created there.
-        .chain(create_indexes.into_iter().map(SqlMigrationStep::CreateIndex))
-        // Order matters: this needs to come after create_indexes, because the foreign keys can depend on unique
-        // indexes created there.
-        .chain(add_foreign_keys.into_iter().map(SqlMigrationStep::AddForeignKey))
-        .chain(alter_indexes.into_iter().map(|idxs| SqlMigrationStep::AlterIndex {
-            table: idxs.as_ref().map(|(table, _)| *table),
-            index: idxs.as_ref().map(|(_, idx)| *idx),
-        }))
-        .chain(
-            redefine_indexes
-                .into_iter()
-                .map(|idxs| SqlMigrationStep::RedefineIndex {
-                    table: idxs.as_ref().map(|(table, _)| *table),
-                    index: idxs.as_ref().map(|(_, idx)| *idx),
-                }),
-        )
-        .collect()
+    steps.extend(
+        flavour
+            .create_enums(&differ)
+            .into_iter()
+            .map(SqlMigrationStep::CreateEnum)
+            .chain(alter_enums.into_iter().map(SqlMigrationStep::AlterEnum))
+            .chain(drop_foreign_keys.into_iter().map(SqlMigrationStep::DropForeignKey))
+            .chain(drop_indexes.into_iter().map(SqlMigrationStep::DropIndex))
+            .chain(alter_tables.into_iter().map(SqlMigrationStep::AlterTable))
+            .chain(
+                drop_tables
+                    .into_iter()
+                    .map(|table_index| SqlMigrationStep::DropTable { table_index }),
+            )
+            .chain(flavour.drop_enums(&differ).into_iter().map(SqlMigrationStep::DropEnum))
+            .chain(redefine_tables)
+            .chain(create_indexes.into_iter().map(SqlMigrationStep::CreateIndex))
+            .chain(add_foreign_keys.into_iter().map(SqlMigrationStep::AddForeignKey))
+            .chain(alter_indexes.into_iter().map(|idxs| SqlMigrationStep::AlterIndex {
+                table: idxs.as_ref().map(|(table, _)| *table),
+                index: idxs.as_ref().map(|(_, idx)| *idx),
+            }))
+            .chain(
+                redefine_indexes
+                    .into_iter()
+                    .map(|idxs| SqlMigrationStep::RedefineIndex {
+                        table: idxs.as_ref().map(|(table, _)| *table),
+                        index: idxs.as_ref().map(|(_, idx)| *idx),
+                    }),
+            ),
```

Comment:
The code is left unchanged there, but we can start simplifying after this PR, now that we don't need to go overboard with iterators anymore.